src/ray/gcs/gcs_client/gcs_client.h

```diff
@@ -61,6 +65,7 @@ class GcsClientOptions {
   // Gcs address
   std::string gcs_address_;
   int gcs_port_ = 0;
+  ClusterID cluster_id_;
```

Comment:
Should we introduce a flag called `fetch_cluster_id_if_not_exist` and only set it for PythonGcsClient. This way the behavior is the same as before: i.e. existing code that uses GcsClient directly without providing cluster id won't fetch cluster id from gcs.

Reply:
eventually we want to have a check that if `fetch_cluster_id_if_not_exists` is false, cluster id must be provided.

Reply:
now C++ GcsClient users (gcs, raylet) are OK with that; But later if we migrate PythonGcsClient users (core_worker, dash) they will have to fetch cluster_id. So I don't think we will ever have such a check.

Reply:
done, added